x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -402,33 +416,39 @@ export const enrichExceptionItemsWithOS = (
   });
 };
 
+/**
+ * Returns the value for the given fieldname within TimelineNonEcsData if it exists
+ */
 export const getMappedNonEcsValue = ({
   data,
   fieldName,
 }: {
   data: TimelineNonEcsData[];
   fieldName: string;
-}): string[] | undefined => {
+}): string[] => {
   const item = data.find((d) => d.field === fieldName);
   if (item != null && item.value != null) {
     return item.value;
   }
-  return undefined;
+  return [];
 };
 
 export const entryHasListType = (
   exceptionItems: Array<ExceptionListItemSchema | CreateExceptionListItemSchema>
 ) => {
   for (const { entries } of exceptionItems) {
     for (const exceptionEntry of entries ?? []) {
```

Comment:
`entries` gets defaulted to empty array, does Typescript show an error without the `?? []` because then I'm wondering if we missed something somewhere.

Reply:
since we type `CreateExceptionListItemSchema` as a potential type within the array, it's doesn't hasn't been defaulted yet

Reply:
Ah, gotcha. Is this being used to check exceptions created within the builder? If that's the case I would type it to `ExceptionsBuilderExceptionItem[]`. That uses `ExceptionListItemBuilderSchema | CreateExceptionListItemBuilderSchema` and `CreateExceptionListItemBuilderSchema` is
```
xport type CreateExceptionListItemBuilderSchema = Omit<
  CreateExceptionListItemSchema,
  'meta' | 'entries'
> & {
  meta: { temporaryUuid: string };
  entries: BuilderEntry[];
};
```

But I guess, that's only if it's being used in the builder.

Reply:
Nah, only being used in both versions of the exception modals so it's usually being passed that `exceptionItemsToAdd` we define with the same `Array<ExceptionListItemSchema | CreateExceptionListItemSchema>` type in component state